gethrpc/client.go

```diff
@@ -480,17 +480,33 @@ func (c *Client) send(ctx context.Context, op *requestOp, msg interface{}) error
 }
 
 func (c *Client) write(ctx context.Context, msg interface{}) error {
+	if err := c.checkWriteConn(ctx); err != nil {
+		return err
+	}
+
+	if err := c.writeConn.Write(ctx, msg); err != nil {
+		c.writeConn = nil
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) checkWriteConn(ctx context.Context) error {
 	// The previous write failed. Try to establish a new connection.
 	if c.writeConn == nil {
-		if err := c.reconnect(ctx); err != nil {
-			return err
-		}
+		return c.reconnect(ctx)
 	}
-	err := c.writeConn.Write(ctx, msg)
-	if err != nil {
-		c.writeConn = nil
+
+	// Extra check to ensure that connection is not closed.
+	select {
```

Comment:
This can be turned into a simple if statement. 

`if c.writeConn.Closed() return err;`

`ctx.Done()` is already being checked in the parent function.

Reply:
They've done the exact same thing as me on the master repo. So I guess it's fine ?

Reply:
Are you saying you copied the code from upstream and created this PR?

Reply:
Yup. Couldn't sync with the upstream cuz there are breaking changes.

Reply:
https://github.com/centrifuge/go-substrate-rpc-client/blob/master/gethrpc/client.go#L501

Reply:
Okay, thank you.

It's fine since I plan on merging this to the main repo after completing the contracts module. I don't wanna deal with unrelated conflicts then.